app/uk/gov/hmrc/incometaxsessiondata/controllers/SessionController.scala

```diff
@@ -71,13 +92,66 @@ class SessionController @Inject() (
+            s" Another document matching mtditid: ${validRequest.mtditid} but different sessionId: ${validRequest.sessionId} and internalId ${validRequest.internalId}"
+        )
+        Future.successful(
+          Forbidden(
+            s"Another document matching mtditid: ${validRequest.mtditid} but different sessionId: ${validRequest.sessionId} and internalId ${validRequest.internalId}"
+          )
+        )
+      case sessionList: List[Session] if sessionList.nonEmpty && isFullDuplicate(sessionList, validRequest)  =>
+        logger.info(
+          s"[SessionController][save]" +
+            s" A session in the database matched the current session request, list of sessions: $sessionList"
+        )
+        handleConflictScenario(validRequest)
+      case sessionList: List[Session] if sessionList.isEmpty                                                 =>
+        logger.info(s"[SessionController][save]: No live sessions matching mtditid: ${validRequest.mtditid}")
+        handleOkScenario(validRequest)
+    }
+
+  private def handleConflictScenario(validRequest: Session) =
     sessionService.set(validRequest) map {
-      case true  =>
-        logger.info(s"[SessionController][save]: Successfully set session")
+      case Right(true)         =>
+        logger.info(
+          s"[SessionController][handleConflictScenario]: Successfully set session despite matching documenting existing in the database"
+        )
+        Conflict("Successfully set session despite matching documenting existing in the database")
+      case Right(false)        =>
+        logger.info(s"[SessionController][handleConflictScenario]: Write operation was not acknowledged")
+        InternalServerError("Write operation was not acknowledged")
+      case Left(ex: Throwable) =>
+        logger.error(
+          s"[SessionController][handleConflictScenario]: Unknown exception < Message: ${ex.getMessage}, Cause: ${ex.getCause} >"
+        )
+        InternalServerError("Unknown exception")
+    }
+
+  private def handleOkScenario(validRequest: Session) =
+    sessionService.set(validRequest) map {
+      case Right(true)         =>
+        logger.info(s"[SessionController][handleOkScenario]: Successfully set session")
         Ok("Successfully set session")
-      case false =>
-        logger.error(s"[SessionController][save]: Failed to set session")
-        InternalServerError("Failed to set session")
+      case Right(false)        =>
+        logger.info(s"[SessionController][handleOkScenario]: Write operation was not acknowledged")
+        InternalServerError("Write operation was not acknowledged")
+      case Left(ex: Throwable) =>
+        logger.error(
+          s"[SessionController][handleOkScenario]: Unknown exception < Message: ${ex.getMessage}, Cause: ${ex.getCause} >"
+        )
+        InternalServerError("Unknown exception")
     }
+
+  private def isFullDuplicate(sessionList: List[Session], validRequest: Session): Boolean = {
```

Comment:
this is really should be a method within a service which returns type of the duplicate not a boolean